### DISCUSSION TITLE
What does isWalking really mean?

### DISCUSSION THREAD

**Discussion by sjaanus** [markus-wa/demoinfocs-golang#317] [General]
I first thought isWalking means that player is in movement. But is this the case?

I went through csgo source code https://github.com/perilouswithadollarsign/cstrike15_src and found this:

`bool walkButtonIsDown = ( mv->m_nButtons & ( /*IN_WALK | */ IN_SPEED ) ) != 0;`

```
if ( walkButtonIsDown )
{
          // we don't cap walk immediately, let the player decelerate and only cap when the speed is really close
          float currentspeed = m_pCSPlayer->GetLocalVelocity( ).Length( );
          if ( currentspeed < ( ( mv->m_flMaxSpeed * CS_PLAYER_SPEED_WALK_MODIFIER ) + 25 ) )
          {
	          mv->m_flMaxSpeed *= CS_PLAYER_SPEED_WALK_MODIFIER;
	          m_pCSPlayer->m_bIsWalking = true;
          }
}
else
{
         m_pCSPlayer->m_bIsWalking = false;
}

```
This makes me think that m_bIsWalking only means that player is holding walk button? This means he can still be standing still.

I have not fully went through the source code, but just an interesting idea. Maybe we are using the flag wrongly?

**Comment by markus-wa** [comment]
interesting find - I'll try to take a look soon.

But it might just be that we do a documentation change as we're surfacing a net-var of the same name, so it might be misleading if it was something else

**Comment by sjaanus** [comment]
I think these are interesting lines also.

It seems like the variable `m_bIsWalking ` is not affecting accuracy, but rather it is calculated by method `RemapValClamped`, based on velocity. Maybe we can get a variable `isMoving` somehow instead of `inWalking`? 

https://github.com/perilouswithadollarsign/cstrike15_src/blob/f82112a2388b841d72cb62ca48ab1846dfcc11c8/game/shared/cstrike15/weapon_csbase.cpp#L1227-L1250

**Comment by sjaanus** [comment]
Confirmed. Just tested on local server. I was standing still holding walk key and shot 4 shots. It said I hit 4 shots and all of them are `isWalking`. We really need a better flag to detect movement. Any suggestions?
